gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/FlowTriggerHandler.java

```diff
@@ -117,13 +117,15 @@ public void handleTriggerEvent(Properties jobProps, DagActionStore.DagAction flo
         this.leaseObtainedCount.inc();
         if (persistFlowAction(leaseObtainedStatus)) {
           log.info("Successfully persisted lease: [{}, eventTimestamp: {}] ", leaseObtainedStatus.getFlowAction(),
-              leaseObtainedStatus.getEventTimestamp());
+              leaseObtainedStatus.getEventTimeMillis());
           return;
         }
         // If persisting the flow action failed, then we set another trigger for this event to occur immediately to
         // re-attempt handling the event
+        DagActionStore.DagAction updatedFlowAction = DagActionStore.DagAction.updateFlowExecutionId(flowAction,
+            leaseObtainedStatus.getEventTimeMillis());
         scheduleReminderForEvent(jobProps,
-            new MultiActiveLeaseArbiter.LeasedToAnotherStatus(flowAction, leaseObtainedStatus.getEventTimestamp(), 0L),
+            new MultiActiveLeaseArbiter.LeasedToAnotherStatus(updatedFlowAction, 0L),
```

Comment:
Do we still need this if we are using the event timetime millisecond, which should be synchronized across all hosts?

Reply:
Yes we do, the timestamp within `leaseObtainedStatus` is the agreed upon time that is synchronized across all hosts while the following param, `eventTimeMillis` (later called) `triggerEventTimeMillis` is local to the host and is only used for logging purposes to show us that we switch from local trigger to the synchronized trigger. Here we are just changing where it's being stored not the fact that we do update it.